src/EditorFeatures/Core/Implementation/IntelliSense/Helpers.cs

```diff
@@ -1,27 +1,85 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Generic;
 using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.Editor.GoToDefinition;
+using Microsoft.CodeAnalysis.Editor.Host;
 using Microsoft.VisualStudio.Text.Adornments;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense
 {
     internal static class Helpers
     {
-        internal static IEnumerable<object> BuildClassifiedTextElements(ImmutableArray<TaggedText> taggedTexts)
+        internal static IEnumerable<object> BuildClassifiedTextElements(ImmutableArray<TaggedText> taggedTexts, Document document, IEnumerable<Lazy<IStreamingFindUsagesPresenter>> streamingPresenters)
+        {
+            var index = 0;
+            return BuildClassifiedTextElements(taggedTexts, ref index, document, streamingPresenters);
```

Comment:
i'm surprised we need presenters to build text elements.

Reply:
They are needed for handling clicks on the text elements.

Reply:
Renaming the method to `BuildInteractiveTextElements` for clarity.